rust/src/dataset/scanner.rs

```diff
@@ -147,6 +147,17 @@ impl<'a> Scanner {
                 Arc::new(projection.clone()),
                 flat_knn_node,
             ))
+        } else if self.limit.is_some() || self.offset.is_some() {
```

Comment:
Could this work with other scanner, i.e., KNN?
for example, should it be the front of the `ExecNode` so far.